app/models/creditsandrefunds/CreditAndRefundViewModel.scala

```diff
@@ -26,33 +25,47 @@ sealed trait CreditRow {
   val creditType: CreditType
 }
 case class CreditViewRow(amount: BigDecimal, creditType: CreditType, taxYear: String) extends CreditRow
-case class PaymentCreditRow(amount: BigDecimal, creditType: CreditType, date: LocalDate) extends CreditRow
+case class PaymentCreditRow(amount: BigDecimal,  date: LocalDate) extends CreditRow {
 
-case class CreditAndRefundViewModel(creditCharges: List[(DocumentDetailWithDueDate, FinancialDetail)]) {
+  override val creditType: CreditType = PaymentType
 
-  val sortedCreditRows: Seq[Option[CreditRow]] = sortCreditsByYear.map(cc => {
+}
+case class RefundRow(amount: BigDecimal) extends CreditRow {
+
+  override val creditType: CreditType = Repayment
+
+}
+
+case class CreditAndRefundViewModel(creditCharges: List[(DocumentDetailWithDueDate, FinancialDetail)], balanceDetails: Option[BalanceDetails]) {
+
+  private val repayments: Seq[CreditRow] = balanceDetails.fold(Seq[RefundRow]())(details =>
+       Seq(details.firstPendingAmountRequested, details.secondPendingAmountRequested)
+        .flatten.sorted(Ordering.BigDecimal.reverse).map(amount => RefundRow(amount)))
+
+  private val creditsAndPayments: Seq[CreditRow] = sortCreditsByYear.flatMap(cc => {
     val (documentDetails, financialDetail) = cc
     val maybeCreditRow = for {
       creditType <- financialDetail.getCreditType
       amount <- documentDetails.documentDetail.paymentOrChargeCredit
     } yield {
       creditType match {
-        case PaymentType => PaymentCreditRow(amount = amount, creditType = creditType, date = documentDetails.dueDate.get)
+        case PaymentType => PaymentCreditRow(amount = amount, date = documentDetails.dueDate.get)
```

Comment:
just wondering if dueDate is guarantied to be present in this context ? ( we are using .get call in line 52)

Reply:
I think we need a refactoring ticket(s) to go through and remove .gets / exceptions on missing fields. Before the refactoring that introduced the credit rows, we were assuming dueDate would be present in this context - an exception was explicitly thrown in the view if it wasn't - https://github.com/hmrc/income-tax-view-change-frontend/blob/d9fc28a20ec474a25546ba7f69b040ff4674b0e5/app/views/CreditAndRefunds.scala.html#L235

Reply:
we already have a story for this work: https://jira.tools.tax.service.gov.uk/browse/MISUV-7424
I hope we are going to continue work on this logic in the next sprint